stdlib/public/SDK/Foundation/Foundation.swift

```diff
@@ -95,3 +95,20 @@ extension CVarArg where Self: _ObjectiveCBridgeable {
     return _encodeBitsAsWords(object)
   }
 }
+
+//===----------------------------------------------------------------------===//
+// Runtime support for NSKeyedArchives
+//===----------------------------------------------------------------------===//
+
+@_silgen_name("swift_registerClassNameForArchiving")
+public func _registerClassNameForArchiving(_ NameForClass: UnsafePointer<CChar>,
+                                           _ Class: AnyClass) {
+  // If it's not possible to create a String from the name, it should abort
+  // and not fail silently.
+  let NameStr = String(validatingUTF8: NameForClass)!
```

Comment:
We do want this to be as little of a performance hit as possible, though. If there's a non-validating version I would prefer that.